### ISSUE TITLE
Don't treat a column as time if none of the rows have a time for that column

### ISSUE THREAD

**Issue by tobybellwood** [TerriaJS/terriajs#2354]
Just come across an example of a dataset that had non-dates in a column, but because the column contains the word "date", terria takes it to be a date column.

note: only true for words starting with date - \b(date)

This data fails to map:

| Lat          | Lon         | Site        | Phase-in date |
|--------------|-------------|-------------|---------------|
| -35.28777551 | 149.1389382 | City        | 2018          |
| -35.23722282 | 149.0913239 | Belconnen   | mid 2017      |
| -35.34784748 | 149.0953871 | Woden       | 2018-19       |
| -35.32633452 | 149.1580553 | Fyshwick    | 2018-2019     |
| -35.41616437 | 149.0707035 | Tuggeranong | late 2020     |
| -35.18639232 | 149.1318297 | Gungahlin   | 2021          |

However, if the column did contain valid dates, the timeslider would start displaying points automatically as if it were a "date" column, which is not the intent of the dataset...

**Comment by RacingTadpole** [comment]
Yes, this is a problem.  However for another csv file, it's possible that you _would_ want the time slider for the phase-in date. 
A good solution would be to present the user with some choices once they add their own data.

Note if it's coming from an init json file, you can use `tableStyle.timeColumn = null` to prevent this.

**Comment by tephenavies** [comment]
The time/date regex is quite permissive, and occasionally it tries to interpret columns that are not time columns. Since TerriaJS [already sorts by the time column](https://github.com/TerriaJS/terriajs/blob/bde847c34/lib/Map/TableStructure.js#L162), it shouldn't be a performance burden to check each row to make sure the column is a time column.

Test csv ([nationalmap link](https://nationalmap.gov.au/#start=%7B%22version%22%3A%220.0.05%22%2C%22initSources%22%3A%5B%7B%0A%22catalog%22%3A%20%5B%7B%20%0A%20%20%22name%22%3A%20%22test%22%2C%0A%20%20%22type%22%3A%20%22csv%22%2C%0A%20%20%22data%22%3A%20%22column%20that%20is%20not%20time%2Cvalue%2Csa4%5Cngreen%2C6%2C101%5Cnblue%2C3%2C102%22%0A%7D%5D%0A%7D%5D%7D)):
```
column that is not time,value,sa4
green,6,101
blue,3,102
```

What happens:
- Column is interpreted as time
- Timeline shown with NaNs because times could not be parsed
- No visualisation shown

What should happen:
- Time column identified
- Time column validation fails
- Visualisation drawn as a region mapped layer with no timeline (and with that column as a potential data variable)
